docs/Catalog/Embed.md

```diff
@@ -0,0 +1,349 @@
+        // ignore messages from other windows
+        e.source !== iframeRef.current.contentWindow ||
+        // ensure origin is what we expect it to be (user has not navigated away)
+        e.origin !== EMBED_ORIGIN ||
+        // ensure the message has expected format (.source set to 'quilt-embed')
+        e.data.source !== EVENT_SOURCE ||
+        // ensure this is "our" instance by comparing nonce passed to the iframe
+        // via query string to the nonce passed back by the iframe
+        nonce !== e.data.nonce
+      ) {
+        return
+      }
+      // handle messages from the Embed
+      switch (e.data.type) {
+        case 'error':
+          console.error(e.data.message, e.data)
+          return
+        case 'ready':
+          initialize()
+          return
+        case 'navigate':
+          // store Embed state (route), e.g. update our URL to store the embed route in a query paramter or smth
+          console.log('embed navigating to', e.data.route)
+          return
+        case 's3ObjectLink':
+          // construct a custom link using e.data and copy it to clipboard or perform any other relevant action
+          console.log('s3 object link clicked', e.data)
+          return
+      }
+    },
+    [iframeRef, nonce, initialize],
+  )
+
+  React.useEffect(() => {
+    // subscribe to messages from Embed
+    window.addEventListener('message', handleMessage)
+    return () => {
+      window.removeEventListener('message', handleMessage)
+    }
+  }, [handleMessage])
+
+  return (
+    <iframe
+      ref={iframeRef}
+      src={src}
+      width="900"
+      height="600"
+      // other things like styling and stuff
+    />
+  )
+}
+```
+
+
+## API reference
+
+### URL and query parameters
+
+The Embed is served off the main catalog server under `/__embed` route
+which takes two optional query parameters:
+
+`nonce` (any unique string) is used to identify "our" Embed instance and make sure we're receiving messages from that same instance.
+
+`origin` must be sent to enable cross-origin message passing from Embed to the parent.
+
+
+### Commands to Embed
+
+Embed accepts commands from the parent via [`postMessage` API](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage).
+Command message data format:
+
+```ts
+interface Command {
+  type: string
+  // ...parameters
+}
+```
+
+Available commands are listed below.
+
+#### `init`
+
+Initialize the Embed.
+This command must be sent after the Embed is ready
+(see `ready` message reference for details).
+
+```ts
+interface InitCommand extends Command {
+  type: 'init'
+
+  // Bucket name, e.g. 'my-bucket'
+  bucket?: string
+  // Path to object or prefix in the given bucket, e.g. 'some-prefix/some-object.csv'
+  path?: string
+
+  // Initial route Embed will navigate to, e.g. '/b/my-bucket/tree/some/path',
+  // takes precedence over bucket / path
+  route?: string
+
+  // Embed accepts any credentials supported by the Quilt authentication endpoint,
```

Comment:
Link to list of support SSO providers https://docs.quiltdata.com/advanced-usage/technical-reference#single-sign-on-sso